lib/src/types.rs

```diff
@@ -53,6 +53,7 @@ pub struct MethodDecl {
 
 #[derive(Debug, Clone, Deserialize, Serialize)]
 pub enum Stmt {
+    // @Decide: What are the Types of While/If?
```

Comment:
I am confused. Didn't we have that defuned already? Should probably be 
```rs
If(Expr, Expr, Expr)?
```
With a comment above explaining what the expressions are as to not confuse the other teams

Reply:
Yes, we had already decided their types, I had just forgotten to remove that comment xd

Reply:
Regarding your other point, the current definition for `If` makes more sense to me, than what you suggested. The first argument is the condition, which can only be an expression. The second is the body of the if, which could be a whole block of statements (and not just an expression). And the third argument is the body of the else statement. It's optional, because there doesn't have to be an else and since it could be a whole block of statements in the body of the else, we need a boxed stmt again.

Reply:
Yeah the current definition is better